clickplc/driver.py

```diff
@@ -201,6 +201,27 @@ class ClickPLC(AsyncioModbusClient):
             return decoder.decode_32bit_float()
         return {f'df{n}': decoder.decode_32bit_float() for n in range(start, end + 1)}
 
+    async def _get_ds(self, start, end):
+        """Read DS registers. Called by `get`.
+
+        DS entries start at Modbus address 00000 (00001 in the Click software's
+        1-indexed notation). Each DS entry takes 16 bits.
+        """
+        if start < 1 or start > 4500:
+            raise ValueError('DS must be in [1, 4500]')
+        if end is not None and (end < 1 or end > 4500):
+            raise ValueError('DS end must be in [1, 4500]')
+
+        address = start - 1
+        count = 2 * (1 if end is None else (end - start + 1))
```

Comment:
Delete the `2 *`. You only need one register per read, so this `2 *` is doubling the communication.

Reply:
Done.